### ISSUE TITLE
Solve performance problem caused by NsDepCopTask being an AppDomainIsolatedTask

### ISSUE THREAD

**Issue by realvizu** [realvizu/NsDepCop#26]
In v1.7 NsDepCopTask's base class was changed from Task to AppDomainIsolatedTask to avoid some esoteric exceptions caused by assembly version clashing with other tasks (see #25).

The downside of this changes is that AppDomainIsolatedTasks are not reused so AppDomain creation, assembly loading and jitting occures every time NsDepCopTask is invoked resulting in a severe performance hit.

So NsDepCopTask should be changed back to derive from Task when possible (when assembly version redirection is no longer needed).

**Comment by realvizu** [comment]
A possible solution could be to run NsDepCop out-of-process, continuously (spin up its own process and let it stay alive) so startup time (which is significant) could be paid only once. NsDepCopTask could be changed back to an in-appdomain task (that is cheap to create) and its sole task would be to pass parameters to the NsDepCop server process and wait for the reply.

**Comment by realvizu** [comment]
NsDepCop.ServiceHost introduced that hosts a dependency analyzer service as a remoting server, communicating on named pipes. Started when needed by an NsDepCopTask and keeps running as long as its parent process (msbuild).
